subworkflows/local/sourmash_profiling.nf

```diff
@@ -1,16 +1,34 @@
 include { SOURMASH_SKETCH                           }   from '../../modules/local/nf-core-modified/sourmash/sketch'
 include { SOURMASH_COMPARE                          }   from '../../modules/local/nf-core-modified/sourmash/compare'
 include { SOURMASH_GATHER                           }   from '../../modules/local/nf-core-modified/sourmash/gather'
+include { SOURMASH_TAXANNOTATE                      }   from '../../modules/local/nf-core-modified/sourmash/taxannotate'
 
 workflow SOURMASH_PROFILING {
     take:
-    sequences  // tuple val(meta), path(assemblies) OR tuple val(meta), path(reads)
-    seqtype
-    // databases  // path(databases)
+    sequences       // tuple val(meta), path(assemblies) OR tuple val(meta), path(reads)
+    seqtype         // val(reads) or val(assemblies)
+    databases_csv  // path(sourmash_dbs_csv) CSV file of database,database_path,lineage_path
 
     main:
     ch_versions = Channel.empty()
 
+    // read in databases from input CSV
+    // collects all of them together to make a list to run once per sample
+    ch_sourmash_databases = Channel.fromPath(databases_csv)
+        .splitCsv(header:true, sep:',')
+        .filter { row -> row.database_path != null }
+        .map { row -> file(row.database_path) }
+        .collect()
```

Comment:
I don't think you need both collect and toList. Just toList may be sufficient.

Reply:
In my testing if you don't collect before setting as a list it doesn't put them together as one so that when you do combine it won't keep the entire list of databases together and gets confused how to input it all

Reply:
If you run `.toList()` by itself the resulting input gather channel looks like: 
`[[id:comm_1_T1, single_end:false], /Users/emcdaniel/Desktop/Arcadia/workflows/metagenomics/work/b6/4cf76eede4b448f83f46fc2892f7db/comm_1_T1.reads.sig, /Users/emcdaniel/Desktop/Arcadia/workflows/test-datasets/metagenomics/sourmash_dbs/GCF_001457635.1.db.zip, /Users/emcdaniel/Desktop/Arcadia/workflows/test-datasets/metagenomics/sourmash_dbs/GCF_003697165.2.db.zip]`

whereas with both `.collect()` and `.toList()` it looks like
`[[id:comm_1_T1, single_end:false], /Users/emcdaniel/Desktop/Arcadia/workflows/metagenomics/work/b6/4cf76eede4b448f83f46fc2892f7db/comm_1_T1.reads.sig, [/Users/emcdaniel/Desktop/Arcadia/workflows/test-datasets/metagenomics/sourmash_dbs/GCF_001457635.1.db.zip, /Users/emcdaniel/Desktop/Arcadia/workflows/test-datasets/metagenomics/sourmash_dbs/GCF_003697165.2.db.zip]]
` where I need the brackets around the list of databases, and for whatever reason just `.toList()` won't get me there. Same with just `.collect()` by itself. Then it works properly running all databases once for each input signature (once per sample)